### PR TITLE
Fix trust parameter not saved and not read

### PR DESCRIPTION
**Pull request by troopa81**
When taking a look at #51110 , I met these issues when dealing with flags :

- when enabling/disabling trust or evaluate default parameter, the project doesn't appeared as modified
- trust parameter is not considered in provider (because we need to read flags before loading layer)

### CODE REVIEW

File: src/core/project/qgsproject.cpp [hunk 1]
```diff
@@ -1936,7 +1946,13 @@ bool QgsProject::readProjectFile( const QString &filename, Qgis::ProjectReadFlag
   profile.switchTask( tr( "Resolving references" ) );
   mRootGroup->resolveReferences( this );
 
-  loadProjectFlags( doc.get() );
+  // we need to migrate old fashion designed QgsSymbolLayerReference to new ones
```
Comment (elpaso) [inline src/core/project/qgsproject.cpp:1949]: is this related?

Comment (troopa81) [reply src/core/project/qgsproject.cpp]: No, that's an unfortunate stash/unstash error. Removed!

**Review by elpaso — APPROVED** [review]
